src/hooking/hooking.hpp

```diff
@@ -195,6 +196,8 @@ namespace big
 		static bool can_create_vehicle();
 
 		static void format_int(int64_t integer_to_format, char* format_string, size_t size_always_64, bool use_commas);
+
+		static void write_physical_script_game_state_data_node(__int64 iface, CPhysicalScriptGameStateDataNode* node);
```

Comment:
could iface be a void?

Reply:
> could iface be a void or void*?

Normal void is not possible (I'm not sure the compiler even allows it), it needs to be a value that holds 64 bits to take the whole value and to pass it completely unchanged to the original function value. `iface` is probably a ptr to a class object however we probably haven't mapped it out yet and don't really need to as we're not modifying it.

Reply:
iface is actually the [this_ptr](https://learn.microsoft.com/en-us/cpp/cpp/this-pointer?view=msvc-170) of the class that this function belongs to, probably CPhysicalObject or something like that. However, since he doesn't need it/care, and is only telling the compiler that the first register used in the __fastcall (all 64-bit functors are __fastcalls) is just some value, and what we really care about is the 2nd register, I feel this is fine. Void is actually not a real type, and void* just tells the compiler the same exact thing, Register 1 is just 64-bits wide.